src/ui/mainwindow.cc

```diff
@@ -1471,37 +1471,45 @@ void MainWindow::wheelEvent( QWheelEvent * ev )
   }
 }
 
+#ifdef Q_OS_MACOS
+// macOS's favourable "close"
+void MainWindow::closeEvent( QCloseEvent * ev )
+{
+  if ( ev->spontaneous() ) {
+    closeAllTabs();
+    showMinimized();
```

Comment:
Needs to replace this with `toggleMainwindow` which needs to add option to `ensure hide/minimized`